### PR TITLE
Make GCP janitor thread-safe

### PR DESCRIPTION
**Pull request by pigboysid**
As discussed in https://github.com/kubernetes-sigs/boskos/issues/142,
the GCP janitor isn't thread-safe: sometimes, it can corrupt the GCloud
config file (which is on-disk, and which is shared by all threads).

Instead of using "gcloud config set X/Y Z", use
"os.environ['CLOUDSDK_X_Y'] = Z".

### CODE REVIEW

**Comment by pigboysid** [comment]
In case it expedites the approval of this review, I'd like to say:

You'll notice that I've removed the checks which validate that the `gcloud config set` commands run without error. My justifications for this are as follows:

- I'm now setting an environment variable rather than running a command, so there's no longer an error to check.
- I tested `gcloud config set billing/quota_project <xyz>` on the command line myself. It appears to do no validation of the project argument `<xyz>` at all: even if that project doesn't exist, it allows you to set it. Thus, it seems to me that the error condition (i.e., `gcloud config set` fails) would never have been triggered.
- Likewise, I tested `gcloud config set api_endpoint_overrides/gkehub`. It _does_ emit an error if the argument isn't of the form `http://<some-url>/` or `https://<some-url>/`, but otherwise, it appears to do no URL validation at all (i.e., it doesn't attempt to connect to the URL you provide). As above, it seems to me that the error condition in the old code would never have been triggered.

File: cmd/janitor/gcp_janitor.py [hunk 1]
```diff
@@ -508,17 +508,7 @@ def activate_service_account(service_account):
...
-        '%s' % project,
-    ]
-    log('running %s' % cmd)
-
-    try:
-        subprocess.check_call(cmd)
-    except subprocess.CalledProcessError:
-        print('Error try to set quota_project: %s' % project, file=sys.stderr)
-        return 1
+    os.environ['CLOUDSDK_BILLING_QUOTA_PROJECT'] = '%s' % project
```
Comment (listx) [inline cmd/janitor/gcp_janitor.py:511]: My Python is rusty so I stackoverflowed and I found https://stackoverflow.com/questions/5971312/how-to-set-environment-variables-in-python#comment61988654_5971326 --- do the concerns there about child processes vs parent process apply here as well? I assume "no" and this is fine, but just wanted to ask.

Comment (pigboysid) [reply cmd/janitor/gcp_janitor.py:511]: Hey Linus: thanks for the prompt attention. To ensure I'm responding to your actual question, you're referring to this comment, yes?

> Caution: to quote from @Evan's reference above, Such changes to the environment affect subprocesses started with os.system(), popen() or fork() and execv(). In other words, keep in mind that this approach won't modify the way your program is running, only the way your program's children run.

In this case, `gcp_janitor.py` *is* the child process (it's spawned from `janitor.go`). So, settings that we make in `gcp_janitor.py` only affect its own specific invocation, not others spawned by `janitor.go`.

You can see that overriding the environment variable *does* affect subsequent `subprocess` invocations in the same script. Some sample code:
```python
import os, subprocess

GCLOUD_CMD = "gcloud config get api_endpoint_overrides/gkehub"

def set_gkehub(api):
  os.environ["CLOUDSDK_API_ENDPOINT_OVERRIDES_GKEHUB"] = f'https://{api}/'

def run_cmd(pstr):
  print(f"-- {pstr}")
  subprocess.run(GCLOUD_CMD, shell=True, text=True, check=True)

run_cmd("No env override")

gkehub_apis = {'gkehub.googleapis.com', 'staging-gkehub.sandbox.googleapis.com', 'autopush-gkehub.sandbox.googleapis.com'}
for api in gkehub_apis:
  set_gkehub(api)
  run_cmd(f"Env override with {api}")
```
And here's the output when I run that code:
```
-- No env override
(unset)
-- Env override with autopush-gkehub.sandbox.googleapis.com
https://autopush-gkehub.sandbox.googleapis.com/
-- Env override with staging-gkehub.sandbox.googleapis.com
https://staging-gkehub.sandbox.googleapis.com/
-- Env override with gkehub.googleapis.com
https://gkehub.googleapis.com/
```
The bottom line is: `os.environ` sets the environment only for a given script and its children, and separate invocations of `gce_janitor.py` (which don't affect each others' environments) are spawned by `janitor.go`.

Comment (pigboysid) [reply cmd/janitor/gcp_janitor.py:511]: Sorry, Linus, I should add one clarification:

Calls to `subprocess` use `popen` under the hood. That's why setting `os.environ` inside my sample code above does affect subsequent calls to `subprocess`.

Comment (listx) [reply cmd/janitor/gcp_janitor.py:511]: > To ensure I'm responding to your actual question, you're referring to this comment, yes?

Yes. (Sorry I didn't make it clear in my original question.)

LGTM

Comment (pigboysid) [reply cmd/janitor/gcp_janitor.py:511]: Thank you, @listx!

By the way, I see a label on this PR saying `needs-ok-to-test`. Are you able to authorize that?
